hs_core/tests/api/native/test_irods_user_zone_federation.py

```diff
@@ -0,0 +1,209 @@
+        test_file.close()
+        test_file = open(self.file_three, 'w')
+        test_file.write("Test text file in file3.txt")
+        test_file.close()
+        test_file = open(self.file_to_be_deleted, 'w')
+        test_file.write("Test text file which will be deleted after resource creation")
+        test_file.close()
+
+        # transfer files to user zone space
+        self.irods_storage = IrodsStorage('federated')
+        irods_target_path = '/' + settings.HS_USER_IRODS_ZONE + '/home/testuser/'
+        self.irods_storage.saveFile(self.file_one, irods_target_path + self.file_one)
+        self.irods_storage.saveFile(self.file_two, irods_target_path + self.file_two)
+        self.irods_storage.saveFile(self.file_three, irods_target_path + self.file_three)
+        self.irods_storage.saveFile(self.file_to_be_deleted,
+                                    irods_target_path + self.file_to_be_deleted)
+
+    def tearDown(self):
+        super(TestUserZoneIRODSFederation, self).tearDown()
+        # no need for further cleanup if federation testing is not setup in the first place
+        if not settings.REMOTE_USE_IRODS or settings.HS_USER_ZONE_HOST != 'users.local.org' \
+                or settings.IRODS_HOST != 'data.local.org':
+            return
+
+        # delete irods test user in user zone
+        super(TestUserZoneIRODSFederation, self).delete_irods_user_in_user_zone()
+
+        os.remove(self.file_one)
+        os.remove(self.file_two)
+        os.remove(self.file_three)
+        os.remove(self.file_to_be_deleted)
+
+    def test_resource_operations_in_user_zone(self):
+        # only do federation testing when REMOTE_USE_IRODS is True and irods docker containers
+        # are set up properly
+        if not settings.REMOTE_USE_IRODS or settings.HS_USER_ZONE_HOST != 'users.local.org' \
+                or settings.IRODS_HOST != 'data.local.org':
+            return
+        # test resource creation and "move" option in federated user zone
+        fed_test_file_full_path = '/{zone}/home/testuser/{fname}'.format(
+            zone=settings.HS_USER_IRODS_ZONE, fname=self.file_to_be_deleted)
+        res = resource.create_resource(
+            resource_type='GenericResource',
+            owner=self.user,
+            title='My Test Generic Resource in User Zone',
+            fed_res_file_names=[fed_test_file_full_path],
+            fed_copy_or_move='move'
+        )
+
+        self.assertEqual(res.files.all().count(), 1,
+                         msg="Number of content files is not equal to 1")
+        fed_path = '/{zone}/home/{user}'.format(zone=settings.HS_USER_IRODS_ZONE,
+                                                user=settings.HS_LOCAL_PROXY_USER_IN_FED_ZONE)
+        user_path = '/{zone}/home/testuser/'.format(zone=settings.HS_USER_IRODS_ZONE)
+        self.assertEqual(res.resource_federation_path, fed_path)
+        # test original file in user test zone is removed after resource creation

```

Comment:
The resource file path is used in subsequent tests/asserts to make sure the files exist or not exist as needed, so we don't need a separate assert here.